framework/auth/__init__.py

```diff
@@ -79,3 +80,24 @@ def register_unconfirmed(username, password, fullname, campaign=None):
     else:
         raise DuplicateEmailError('User {0!r} already exists'.format(username))
     return user
+
+
+def get_or_create_user(fullname, address, is_spam=False):
+    """Get or create user by email address.
+
+    :param str fullname: User full name
+    :param str address: User email address
+    :param bool is_spam: User flagged as potential spam
+    :return: Tuple of (user, created)
+    """
+    user = get_user(email=address)
+    if user:
+        return user, False
+    else:
+        from website import security  # Avoid circular imports
+        password = str(uuid.uuid4())
+        user = User.create_confirmed(address, password, fullname)
+        user.verification_key = security.random_string(20)
```

Comment:
Oh, is the user expected to reset their password after their account is created?